mlrun/run.py

```diff
@@ -1256,716 +1268,161 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
-
-        :returns: The `DataItem` as a `dict`.
-        """
-        return data_item.as_df().to_dict()
-
-    @staticmethod
-    def parse_list(data_item: DataItem) -> list:
-        """
-        Parse an MLRun `DataItem` to a `list`.
-
-        :param data_item: The `DataItem` to parse.
-
-        :returns: The `DataItem` as a `list`.
-        """
-        return data_item.as_df().to_numpy().tolist()
-
-    @staticmethod
-    def parse_object(data_item: DataItem) -> object:
-        """
-        Parse an MLRun `DataItem` to its unpickled object. The pickle file will be downloaded to a local temp
-        directory and then loaded.
-
-        :param data_item: The `DataItem` to parse.
-
-        :returns: The `DataItem` as the original object that was pickled once it was logged.
-        """
-        object_file = data_item.local()
-        with open(object_file, "rb") as pickle_file:
-            obj = cloudpickle.load(pickle_file)
-        return obj
-
-
-class OutputsLogger:
-    """
-    A static class to hold all the common logging functions - functions for logging different objects by artifact type
-    to MLRun.
-    """
-
-    @staticmethod
-    def log_dataset(
-        ctx: MLClientCtx,
-        obj: Union[pd.DataFrame, np.ndarray, pd.Series, dict, list],
-        key: str,
-        logging_kwargs: dict,
-    ):
-        """
-        Log an object as a dataset. The dataset wil lbe cast to a `pandas.DataFrame`. Supporting casting from
-        `pandas.Series`, `numpy.ndarray`, `dict` and `list`.
-
-        :param ctx:            The MLRun context to log with.
-        :param obj:            The data to log.
-        :param key:            The key of the artifact.
-        :param logging_kwargs: Additional keyword arguments to pass to the `context.log_dataset`
-
-        :raise MLRunInvalidArgumentError: If the type is not supported for being cast to `pandas.DataFrame`.
-        """
-        # Check for the object type:
-        if not isinstance(obj, pd.DataFrame):
-            if isinstance(obj, (np.ndarray, pd.Series, dict, list)):
-                obj = pd.DataFrame(obj)
-            else:
-                raise mlrun.errors.MLRunInvalidArgumentError(
-                    f"The value requested to be logged as a dataset artifact is of type '{type(obj)}' and it "
-                    f"cannot be logged as a dataset. Please parse it in your code into one `numpy.ndarray`, "
-                    f"`pandas.DataFrame`, `pandas.Series`, `dict`, `list` before returning it so we can log it."
-                )
-
-        # Log the DataFrame object as a dataset:
-        ctx.log_dataset(**logging_kwargs, key=key, df=obj)
-
-    @staticmethod
-    def log_directory(
-        ctx: MLClientCtx,
-        obj: Union[str, Path],
-        key: str,
-        logging_kwargs: dict,
+    # Validate the type hint is a valid module path:
+    if not bool(
+        re.fullmatch(r"([a-zA-Z_][a-zA-Z0-9_]*\.)*[a-zA-Z_][a-zA-Z0-9_]*", type_hint)
```

Comment:
what exactly you are trying to catch here?

Reply:
See comment a line above:
```
# Validate the type hint is a valid module path:
```
A valid module path is like so `module.inner_module.SomeType` so one can `import module.inner_module` or `from module.inner_module import SomeType`.

If a user gave an incorrect module path, its for sure an incorrect type hint.